data-prepper-api/src/main/java/com/amazon/dataprepper/model/span/Span.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.dataprepper.model.span;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A span event in Data Prepper represents a single operation in a trace. Multiple spans can represent a single trace or a service map.
+ */
+public interface Span extends Event {
+
+    /**
+     * Gets an id for the trace as a HexString.
+     * @return the trace id
+     */
+    String getTraceId();
+
+    /**
+     * Gets the id for the span as a HexString.
+     * @return the span id
+     */
+    String getSpanId();
+
+    /**
+     * Gets the trace_state.
+     * @return the trace state
+     */
+    String getTraceState();
+
+    /**
+     * Gets the span id of the parent as a HexString.
+     * @return the parent span id
+     */
+    String getParentSpanId();
+
+    /**
+     * Gets a String description of the span's operation.
+     * @return the name
+     */
+    String getName();
+
+    /**
+     * Gets the string representation of the span type.
+     * @return the kind
+     */
+    String getKind();
+
+    /**
+     * Gets ISO8601 representation of the start time.
+     * @return the start time
+     */
+    String getStartTime();
```

Comment:
Another example of where we may be able to use `Instant` or `OffsetDateTime`.

Reply:
I follow that. But, do we want to make our interface more abstract for consumers of it? 

This could get into a higher-level question of what "primitive" types do we want to have in all of our models and thus in pipeline configurations.

Reply:
Sorry, I didn’t refresh my page before publishing my last comment so I missed your reply.

I am not sure I understand how this will benefit the user by making these durations / times more abstract when they are received by DP as a string and written to OS as a string and we perform a very minimal processing. It may simplify building a prepper. One option would be to have the underlying data structure still be a String with the ISO8601 representation that is translated to an `Instant` for this retrieval method. That would help support operating on this object generically but create more usable return type. 

However, I propose we stick with the current proposed solution. If we feel making the `getStartTime` and `getDuration methods return a more abstract method we can extend this interface in the future to have additional ways to retrieve the `startTime` or `duration`.